### PR TITLE
Async Client communications

### PR DESCRIPTION
**Pull request by grondinjc**

### CODE REVIEW

**Comment by grondinjc** [comment]
@vphilippon Ready for review

File: src/static/js/apps/ide.js [hunk 1]
```diff
@@ -1,175 +1,413 @@
...
+  });
+  tree.initRoot("tree");
 
-  this._difftool = null;
+  // Handle ways of sending and receiving data from/to server
+  var onopen = function(){ obj._ideState.init(); };
+  var onclose = function(){ /* For future usage */ };
+  var onreceive = function(opCode, jsonObj) { obj.handleReceive(opCode, jsonObj); };
+  var requestHandler = new RequestHandler('ide', onreceive, onopen, onclose);
+  // TODO block / wait for the WS to be open before allowing ws-based stuff
```
Comment (vphilippon) [inline src/static/js/apps/ide.js:21]: We can remove this TODO now.

File: src/static/js/apps/ide.js [hunk 2]
```diff
@@ -1,175 +1,413 @@
...
+
+  // debug
+  this._debugStopWatchLap.show();
+};
+IdeEditState.prototype.leave = function(){
+  // Avoid sending other packages
+  clearInterval(this._pushIntervalHandle);
+  // Make sure no changes are left behind
+  this._pushChanges();
+  // Unregister to file ... call /close
```
Comment (vphilippon) [inline src/static/js/apps/ide.js:263]: Should we implement the call to /close now?

Comment (grondinjc) [inline src/static/js/apps/ide.js:263]: yes, feel free to do it

Comment (vphilippon) [inline src/static/js/apps/ide.js:263]: I'll do it after the merge then, next PR

**Comment by vphilippon** [comment]
:+1:

**Pull request merged**